Backport of:

From b61b7f4f0e705b6a9a9ba8b8af898a406b0fc87e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bastien=20ROUCARI=C3=88S?= <roucaries.bastien@gmail.com>
Date: Fri, 7 Nov 2014 21:05:07 +0100
Subject: [PATCH] Avoid crash and DOS with special crafted jpeg file

Some special crafted JPEG file could lead to dos due to missing check in
embeded EXIF properties (EXIF directory offsets must be greater than 0).

Fix CVE-2014-8716.

Forwarded: http://www.imagemagick.org/discourse-server/viewtopic.php?f=3&t=26456
Bug-debian: http://bugs.debian.org/768494
Applied-Upstream: 6.9.9.10
---
 magick/property.c | 2 ++
 1 file changed, 2 insertions(+)

Index: imagemagick-6.6.9.7/magick/property.c
===================================================================
--- imagemagick-6.6.9.7.orig/magick/property.c	2016-11-14 14:10:44.041843061 -0500
+++ imagemagick-6.6.9.7/magick/property.c	2016-11-14 14:11:15.933844548 -0500
@@ -1316,6 +1316,8 @@
             The directory entry contains an offset.
           */
           offset=(ssize_t) ((int) ReadPropertyLong(endian,q+8));
+          if ((offset < 0) || (size_t) offset >= length)
+            continue;
           if ((offset+number_bytes) < offset)
             continue;  /* prevent overflow */
           if ((size_t) (offset+number_bytes) > length)
